static/scripts/assignment_table.js

```diff
@@ -0,0 +1,481 @@
+                    colspan: 1
+                });
+            }
+        }
+
+        const table = new Handsontable(document.getElementById("handsontable"), {
+            data: data,
+            fixedColumnsLeft: lenFixedHeaders,
+            fixedRowsTop: lenFixedRowsText,
+            manualColumnMove: true,
+            manualColumnResize: true,
+            manualRowResize: true,
+            mergeCells: mergeCellsSettings,
+            hiddenColumns: {
+                columns: [0],
+                indicators: true,
+            },
+            hiddenRows: {
+                rows: [0],
+                indicators: true,
+            },
+            contextMenu: ['commentsAddEdit', 'commentsRemove', 'hidden_columns_hide', 'hidden_rows_hide', 'hidden_columns_show', 'hidden_rows_show'],
+            comments: true,
+            filters: true,
+            dropdownMenu: ['filter_by_value', 'filter_action_bar', 'undo'],
+            className: 'controlsQuickFilter htCenter htMiddle',
+            colHeaders: allHeaders,
+            columns: columns,
+            colWidths: 100,
+            columnHeaderHeight: 225,
+            rowHeaders: fixedRowsText,
+            rowHeaderWidth: 125,
+            // Disable the cells that contain user preferences
+            cells: function (row, col) {
+                const cellProperties = {};
+                if (row >= lenFixedRowsText && col >= lenFixedHeaders && row % 2 === 0) {
+                    cellProperties.readOnly = true;
+                }
+                return cellProperties;
+            },
+            afterInit: function () {
+                const sourceData = this.getSourceData();
+                saved_data.forEach(assignment => {
+                    const rowIndex = sourceData.findIndex(row => row.researchers.id === assignment.researcher_id);
+
+                    if (rowIndex !== -1) {
+                        const colIndex = this.propToCol(assignment.course_id);
+
+                        const commentsPlugin = this.getPlugin('comments');
+                        //Id is set for the two merged rows, and we want to set the comment for the assignment row
+                        commentsPlugin.setCommentAtCell(rowIndex + 1, colIndex, assignment.comment);
+                    }
+                });
+            },
+            afterGetColHeader: function (col, th) {
+                th.style.transform = 'rotate(180deg)';
+                th.style.writingMode = 'vertical-lr';
+                th.style.textAlign = 'center';
+                th.style.fontWeight = 'bold';
+                th.style.height = '100px';
+                th.style.lineHeight = '100px';
+
+                if (col > lenFixedHeaders - 1) {
+                    let colData = this.getDataAtCol(col);
+                    if (colData[RowIndices.TOTAL_ASSISTANT_NOW] >= colData[RowIndices.ASSISTANTS]) {
+                        th.style.backgroundColor = 'green';
+                    }
+                }
```

Comment:
Your check on the number of assignations per course is not complete because you only check if that number is lower or equal to the desired value. You don't check if it goes over (red).

Also, you don't do that check if there is no desired number of assigned researcher. The title remains green all the time. I don't know what is best : giving a number at course creation or just leaving the title white if no number is given. What do you think @anthonygego @luciledierckx

Reply:
If there is no desired number of researchers assigned, the value is set to 0 by default. Would you like me to set the header to ‘’white‘’ if the value is the default?

Reply:
I feel that in the normal use of the application, we should define a number of researchers per course. Obviously a person creating a course might not now exactly how many are needed at first. And that is why we handle the case where none are given. 

I was thinking we might use a way to clearly indicate that desired number is not defined for a course. For example, ith a title in light purple or something similar. It would be a way of saying "Be advised, you didn't define a desired number of researcher !".
That additional color would prevent the case where we confuse the courses that are not fully assigned and those with no desired number of researchers.

I will discuss this possibility with the others.